builds/e2e/upgrade-scenarios.yaml

```diff
@@ -0,0 +1,229 @@
+          sed -i 's/"edgeAgentImage.*/"edgeAgentImage": "edgebuilds.azurecr.io\/microsoft\/azureiotedge-agent:'${imageId}'",/' context.json
+          sed -i 's/"edgeHubImage.*/"edgeHubImage": "edgebuilds.azurecr.io\/microsoft\/azureiotedge-hub:'${imageId}'",/' context.json
+        fi
+        if [ $(bootstrapAgentVersion) = "release/1.1" ]; then
+          sed -i 's/"edgeAgentBootstrapImage.*/"edgeAgentBootstrapImage": "edgebuilds.azurecr.io\/microsoft\/azureiotedge-agent:'${imageId}'",/' context.json
+        fi
+        echo "Updated context.json:"        
+        less context.json
+      displayName: Update context.json w/ release/1.1 images 
+
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Release/1.2 Images
+      inputs:
+        buildType: specific
+        project: $(resources.pipeline.images_release_1_2.projectID)
+        pipeline: $(resources.pipeline.images_release_1_2.pipelineName)
+        buildVersionToDownload: specific
+        buildId: $(resources.pipeline.images_release_1_2.runID)
+        downloadType: single
+        artifactName: $(az.pipeline.images.artifacts)
+        itemPattern: $(az.pipeline.images.artifacts)/artifactInfo.txt
+
+    # If $runtimeModulesVersion or $bootstrapAgentVersion is set to 'release/1.2', retrieve
+    # image ID from artifactInfo.txt and update context.json with it.
+    - bash: |
+        artifactInfoStr=`cat '$(System.ArtifactsDirectory)/$(az.pipeline.images.artifacts)/artifactInfo.txt'`
+        IFS='='
+        read -ra artifactInfo <<< "$artifactInfoStr"
+        imageId="${artifactInfo[1]}"
+        echo "imageId: ${imageId}"
+        cd $(Build.SourcesDirectory)/test/Microsoft.Azure.Devices.Edge.Test/bin/Debug/net*
+        echo "Original context.json:"
+        less context.json
+        if [ $(runtimeModulesVersion) = "release/1.2" ]; then
+          sed -i 's/"edgeAgentImage.*/"edgeAgentImage": "edgebuilds.azurecr.io\/microsoft\/azureiotedge-agent:'${imageId}'",/' context.json
+          sed -i 's/"edgeHubImage.*/"edgeHubImage": "edgebuilds.azurecr.io\/microsoft\/azureiotedge-hub:'${imageId}'",/' context.json
+        fi
+        if [ $(bootstrapAgentVersion) = "release/1.2" ]; then
+          sed -i 's/"edgeAgentBootstrapImage.*/"edgeAgentBootstrapImage": "edgebuilds.azurecr.io\/microsoft\/azureiotedge-agent:'${imageId}'",/' context.json
+        fi
+        echo "Updated context.json:"        
+        less context.json
+      displayName: Update context.json w/ release/1.2 images
+
+    # If $runtimeModulesVersion or $bootstrapAgentVersion is set to 1.x.y,
+    # run sed command to set edgeAgentBootstrapImage, edgeAgentImage, and edgeHubImage
+    # to released images from mcr.microsoft.com
+    - bash: |
+        cd $(Build.SourcesDirectory)/test/Microsoft.Azure.Devices.Edge.Test/bin/Debug/net*
+        echo "Original context.json:"
+        less context.json
+        if [ `echo $(runtimeModulesVersion) | sed -En "/1.[0-9]+.[0-9]+/p"` ]; then
```

Comment:
Hmm... this doesn't handle case of Rc.

Reply:
Same response here - let's leave this for a future PR if/when we need it.